http/jetty12/src/main/java/org/apache/felix/http/jetty/internal/JettyService.java

```diff
@@ -253,9 +253,12 @@ private void initializeJetty() throws Exception
             loginService.setUserStore(new UserStore());
             this.server.addBean(loginService);
 
-            ServletContextHandler context = new ServletContextHandler(this.config.getContextPath(),
+            ServletContextHandler context = new ServletContextHandler(this.config.getContextPath(),                    
                     ServletContextHandler.SESSIONS);
 
+            // To support WebSockets via https://github.com/apache/felix-dev/pull/310
+            context.setCrossContextDispatchSupported(true);
```

Comment:
I wonder if there is some way that the (JETTY_WEBSOCKET_CONTAINER_ATTRIBUTE and/or ServerContainer.class.getName()) request attribute could be copied into to (or re-routed from) the  SharedServletContextImpl object that is the delegatee for all of the PerBundleServletContextImpl objects.

If that is possible and works then we wouldn't have to mess with the cross context dispatch flag.  I am not sure what the implications or side effects of that might be.  Plus we would be able to lookup the websocket container directly from the PerBundleServletContextImpl and not have to switch to the root context which seems like an implementation detail that the downstream developers shouldn't need to know.

Reply:
From what i can find, cross context support allows for dospatching requests between different Jetty contexts, as can be seen in https://github.com/jetty/jetty.project/issues/10617.

For Felix HTTP there is only one context, so i would assume there are no side-effects, but i'm not sure of this.

I agree that having a proper alternative would be better though.

Reply:
@enapps-enorman i fixed it like this, what do you think?
https://github.com/apache/felix-dev/pull/310/commits/38b99d7b799a3712374d5f5b3608359bb3d57626

I will work on adapting the example code as well.